conda/common/io.py

```diff
@@ -261,6 +261,94 @@ class CapturedText(object):
         log.info("stderr and stdout yielding back")
 
 
+class CapturedDescriptor:
```

Comment:
We could move this to `conda-libmamba-solver` if needed. It wraps a raw file descriptor and saves it in a buffer. Then we forward that to the logger. We only use it to wrap `libsolv` debug lines, but it might be useful for other compiled libraries in the future?

Up to you :)

Reply:
Let's move it to CLS for now since I'm not sure if we should it at this point tbh.